### PR TITLE
Fixed processing of errors on the client (legacy API, closes #660)

### LINKED ISSUE #660: Failed test displayed as success in reports, if option `screenshotsOnFails` is enabled (legacy)
It can be reproduced via CLI with arguments `-S -s <screenshot path>`
```
'@page http://example.com';
'@fixture failing';


'@test'['Simple falling test'] = {
    '0.Step': function () {
        ok(false);
    }
};
```

It happens because client sends `done` command earlier than `assertion-failed`.

### PR DESCRIPTION
**Pull request by georgiy-abbasov**
\cc @AlexanderMoskovkin, @inikulin, @VasilyStrelyaev

### CODE REVIEW

**Comment by inikulin** [comment]
Is this affects only legacy API? If so, can you add remark both to PR and issue titles?

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

File: test/client/fixtures/runner/runner-test.js [hunk 1]
```diff
@@ -111,3 +114,36 @@ asyncTest('Test iterator should not call Transport.fail twice (with screenshots)
         start();
     }, 100);
 });
+
+asyncTest("The 'assertion-failed' command should be sended earlier than 'done' if the option 'takeScreenshotOnFail' is enabled (GH-660)", function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/runner-test.js]:
```
asyncTest("The 'assertion-failed' command should be sent earlier than 'done' if the 'takeScreenshotOnFail' option is enabled (GH-660)", function () {
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by georgiy-abbasov** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
@georgiy-abbasov What about 

> Is this affects only legacy API? If so, can you add remark both to PR and issue titles?

?

**Comment by georgiy-abbasov** [comment]
@AlexanderMoskovkin, my mistake, forgot to update title of this PR and commit message.

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by georgiy-abbasov** [comment]
ping @AlexanderMoskovkin

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by georgiy-abbasov** [comment]
ping @inikulin

**Comment by inikulin** [comment]
lgtm

**Pull request merged**